entity-store/src/main/kotlin/jetbrains/exodus/entitystore/youtrackdb/iterate/YTDBQueryEntityIterator.kt

```diff
@@ -66,7 +66,7 @@ class YTDBQueryEntityIterator(private val source: Iterator<Entity>, private val
     }
 
     override fun shouldBeDisposed(): Boolean {
-        return false
+        return true
```

Comment:
Good catch @lpld, hopefully it will fix most of the issues, but it would be better to be sure of it. So let us discuss measures on the YTDB level too.